Controllers/FamilyController.cs

```diff
@@ -139,47 +241,202 @@ public async Task<IActionResult> CreateFamily([FromForm] FamilyViewModel Family)
+            }
+
+            // Create all new considerations for update
+            foreach (SelectListItem r in Member.Restrictions)
+            {
+                if (r.Selected && contextMember != null)
+                {
+                    ConsiderationsCreateDto restriction =
+                        new()
+                        {
+                            MemberId = contextMember.MemberId,
+                            Type = ConsiderationsEnum.Restriction,
+                            Value = r.Text
+                        };
+                    var created = _considerationsService.CreateConsideration(restriction);
+                    if (!created.Success)
+                    {
+                        return Task.FromException(
+                            new Exception($"Error creating consideration. Error: {created.Error}")
+                        );
+                    }
+                }
+            }
+
+            foreach (SelectListItem g in Member.Goals)
+            {
+                if (g.Selected && contextMember != null)
+                {
+                    ConsiderationsCreateDto goal =
+                        new()
+                        {
+                            MemberId = contextMember.MemberId,
+                            Type = ConsiderationsEnum.Goal,
+                            Value = g.Text
+                        };
+                    var created = _considerationsService.CreateConsideration(goal);
+                    if (!created.Success)
+                    {
+                        return Task.FromException(
+                            new Exception($"Error creating consideration. Error: {created.Error}")
+                        );
+                    }
+                }
+            }
+
+            foreach (SelectListItem c in Member.Cuisines)
+            {
+                if (c.Selected && contextMember != null)
+                {
+                    ConsiderationsCreateDto cuisine =
+                        new()
+                        {
+                            MemberId = contextMember.MemberId,
+                            Type = ConsiderationsEnum.Cuisine,
+                            Value = c.Text
+                        };
+                    var created = _considerationsService.CreateConsideration(cuisine);
+                    if (!created.Success)
+                    {
+                        return Task.FromException(
+                            new Exception($"Error creating consideration. Error: {created.Error}")
+                        );
+                    }
+                }
+            }
+            contextMember = null;
+        }
+
+        return Task.CompletedTask;
     }
 
-    [HttpPut("{Id}")]
-    public ActionResult<FamilyModel> UpdateFamily(string Id, [FromBody] FamilyUpdateDto family)
+    // Handles the deletion of considerations if a member was to update theirs
+    private Task DeleteOldConsiderations(string memberId)
```

Comment:
could this be moved to the consideration service?

Reply:
Yep, good call!